dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -704,11 +710,28 @@ ALWAYS_INLINE void Aggregator::executeImplBatch(
         AggregateDataPtr place = aggregates_pool->alloc(0);
         for (size_t i = 0; i < agg_size; ++i)
         {
-            auto emplace_result_hold
-                = emplaceKey(method, state, agg_process_info.start_row, *aggregates_pool, sort_key_containers);
+            auto emplace_result_hold = emplaceOrFindKey<only_lookup>(
+                method,
+                state,
+                agg_process_info.start_row,
+                *aggregates_pool,
+                sort_key_containers);
             if likely (emplace_result_hold.has_value())
             {
-                emplace_result_hold.value().setMapped(place);
+                if constexpr (collect_hit_rate)
+                {
+                    ++agg_process_info.hit_row_cnt;
+                }
+
+                if constexpr (only_lookup)
+                {
+                    if (!emplace_result_hold.value().isFound())
+                        agg_process_info.not_found_rows.push_back(i);
```

Comment:
need to setMapped if `emplace_result_hold.value().isFound()` is true?

Reply:
yes.

Reply:
I think no need to setMapped even if found this key. Because if found, the AggregateDataPtr has already setup correctly in HashMap.

PS: the `place` pointer is not used when there is no agg func, I think we can delete setMapped directly?